src/Cli/dotnet/commands/dotnet-workload/search/WorkloadSearchCommand.cs

```diff
@@ -34,10 +41,43 @@ public WorkloadSearchCommand(
 
             _sdkVersion = creationResult.SdkVersion;
             _workloadResolver = creationResult.WorkloadResolver;
+
+            _numberOfWorkloadSetsToTake = result.GetValue(SearchWorkloadSetsParser.TakeOption);
+            _workloadSetOutputFormat = result.GetValue(SearchWorkloadSetsParser.FormatOption);
+
+            _installer = WorkloadInstallerFactory.GetWorkloadInstaller(
+                reporter,
+                new SdkFeatureBand(_sdkVersion),
+                _workloadResolver,
+                Verbosity,
+                creationResult.UserProfileDir,
+                !SignCheck.IsDotNetSigned(),
+                restoreActionConfig: new RestoreActionConfig(result.HasOption(SharedOptions.InteractiveOption)),
+                elevationRequired: false,
+                shouldLog: false);
         }
 
         public override int Execute()
         {
+            if (ListWorkloadSetVersions)
+            {
+                var featureBand = new SdkFeatureBand(_sdkVersion);
+                var packageId = _installer.GetManifestPackageId(new ManifestId("Microsoft.NET.Workloads"), featureBand);
+                var versions = PackageDownloader.GetLatestPackageVersions(packageId, _numberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: !string.IsNullOrWhiteSpace(_sdkVersion.Prerelease))
+                    .GetAwaiter().GetResult()
+                    .Select(version => WorkloadManifestUpdater.WorkloadSetPackageVersionToWorkloadSetVersion(featureBand, version.Version.ToString()));
+                if (_workloadSetOutputFormat?.Equals("json", StringComparison.OrdinalIgnoreCase) == true)
+                {
+                    Reporter.WriteLine(JsonSerializer.Serialize(versions));
+                }
+                else
+                {
+                    Reporter.WriteLine(string.Join(',', versions));
```

Comment:
probably need to add a space after the comma, so `", "`

Reply:
I was a little split on this, actually. Visually, I certainly agree. The other format I created is clearly intended for parsing, and I thought making it a true csv would make this version easy for parsing in an alternate scheme. If you don't think that's necessary (because json already exists), I can do that.